admin/model/localisation/language.php

```diff
@@ -255,20 +218,139 @@ public function addLanguage($data) {
 
         $this->cache->delete('menu');
 
-        // Email Template
-        $query = $this->db->query("SELECT * FROM " . DB_PREFIX . "email_description WHERE language_id = '" . (int)$this->config->get('config_language_id') . "'");
-
-        foreach ($query->rows as $email) {
-            $this->db->query("INSERT INTO " . DB_PREFIX . "email_description SET email_id = '" . (int)$email['email_id'] . "', name = '" . $this->db->escape($email['name']) . "', description = '" . $this->db->escape($email['description']) . "', status = '1', language_id = '" . (int)$language_id . "'");
-        }
+        // Insert Email templates, order statuses, stock statuses, return statuses and return reasons languages
+        $this->prepareLanguages($language_id, $this->db);
 
         $this->cache->delete('email_template');
+        $this->cache->delete('order_status');
+        $this->cache->delete('stock_status');
+        $this->cache->delete('return_status');
+        $this->cache->delete('return_reason');
+        $this->cache->delete('return_action');
 
         $this->trigger->fire('post.admin.language.add', array(&$language_id));
 
         return $language_id;
     }
 
+    public function prepareLanguages($language_id, $db)
+    {
+        $this->language->load('email_template');
+        $this->language->load('order_status');
+        $this->language->load('stock_status');
+        $this->language->load('return_status');
+        $this->language->load('return_reason');
+        $this->language->load('return_action');
+
+        $data = $this->language->all();
+        $this->emailTemplateLanguages($data, $language_id, $db);
+        $this->orderStatusLanguages($data, $language_id, $db);
+        $this->stockStatusLanguages($data, $language_id, $db);
+        $this->returnStatusLanguages($data, $language_id, $db);
+        $this->returnReasonLanguages($data, $language_id, $db);
+        $this->returnActionLanguages($data, $language_id, $db);
+    }
+
+    private function emailTemplateLanguages($data, $language_id, $db)
+    {
+        $finder = new Finder();
+        $finder->files()->in(DIR_ADMIN . 'view/template/email');
+        foreach ($finder as $email_template)
+        {
+            $email_template_id = rtrim($email_template->getFilename(), '.' . $email_template->getExtension());
+            $item              = explode('_', $email_template_id);
+            $query             = $db->query("SELECT id FROM " . DB_PREFIX . "email WHERE type = '" . $item[0] . "' AND text_id = " . $item[1]);
+
+            $content = $this->load->view('email/' . $email_template->getFilename(), $data);
```

Comment:
`view` method `extract`'s the `$data` array as variables so shouldn't we use them as `$xyz` indead of `$data['xyz']` in `.tpl` files?